### ISSUE TITLE
isso loads on webpage, unable to post comments

### ISSUE THREAD

**Issue by lmburns** [posativ/isso#710]
#### Problem
I am able to get `isso` to load on the webpage, as well as have acess to the admin page; however, the comment/post button does not do anything. The preview button works and so does everything else except for actually being able to post a comment.

#### Operating System
- Operating System: Debian GNU/Linux bullseye/sid
- Kernel: Linux 5.10.0-3-amd64

#### Webpage server
- Nginx

I have followed the documentation and whenever that didn't work I tried several websites that have tutorials.

First I tried `pip install isso` and installation went fine. I created 3 different `systemd` services and all resulted in the same thing.

I then tried creating an isso user and using `virutalenv`, and `gunicorn`.  I tried creating the user in the following ways:
```sh
$ adduser --disabled-login --gecos 'Isso Comments' isso
$ adduser isso --disabled-login --disabled-password
$ adduser isso --disabled-login
$ adduser isso
```

I changed permssions on the log file and the database files, but still the same result.

Every time running both:
```sh
$ curl localhost:8080/admin
$ curl https://isso.burnsac.xyz/admin
```
worked just fine and returned an HTML document.

I tried embedding the javascript into a `<div>`, `<section>`, and without anything, while trying all three with and without `type='text/javascript'` tags.

I modified my `isso.cfg` to be the bare minimum as well as having a bunch of options modified.

Finally, I tried installing the native Debian package and used the service that it came with, as well as modifying my `nginx` configuration file to http

#### `isso.cfg`

```
[general]
name = burnsac.xyz
dbpath = /var/lib/isso/com.db
host =
  http://burnsac.xyz
  https://burnsac.xyz
max-age = 15m
notify = stdout
log-file = /var/log/isso.log
gravatar = false
reply-notifications = false

[moderation]
enabled = false

[admin]
enabled = true
password = <password>

[guard]
enabled = true
ratelimit = 2
direct-reply = 3
reply-to-self = false
require-author = false
require-email = false

[smtp]
username = isso
password = <password>
host = localhost
port = 25
security = none
to = <to@email>
from = isso@burnsac.xyz
timeout = 10
```

#### `/etc/nginx/sites-enabled/isso` - An instance of `isso` where I would be redirecting/hosting comments

```
server {
	server_name isso.burnsac.xyz www.isso.burnsac.xyz;

  add_header Strict-Transport-Security max-age=63072000;
  add_header X-Frame-Options DENY;
  add_header X-Content-Type-Options nosniff;
  # Prevent XSS
  add_header X-XSS-Protection "1; mode=block";
  # Set Content Security Policy to restrict JS.
  add_header Content-Security-Policy "script-src isso.burnsac.xyz";
  # Only pass referrer information over HTTPS
  add_header Referrer-Policy no-referrer-when-downgrade;

  # Do not advertise nginx version
  server_tokens off;

  location / {
    proxy_pass http://127.0.0.1:8000;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-Proto $scheme;
  }

    listen [::]:443 ssl http2; # managed by Certbot
    listen 443 ssl http2; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/isso.burnsac.xyz/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/isso.burnsac.xyz/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

    access_log /var/log/nginx/isso-access.log;
    error_log /var/log/nginx/isso-error.log;

}

server {
    if ($host = www.isso.burnsac.xyz) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    if ($host = isso.burnsac.xyz) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

	listen 80 ;
	listen [::]:80 ;

	server_name isso.burnsac.xyz www.isso.burnsac.xyz;
    return 404; # managed by Certbot
}
```

#### `/etc/nginx/sites-enabled/burnsac` - The main website where the comments would be held

```
server {

	root /var/www/burnsac;

	index index.html index.htm index.nginx-debian.html;

	server_name burnsac.xyz www.burnsac.xyz;

	location / {
		try_files $uri $uri/ =404;
	}

    add_header Content-Security-Policy "script-src isso.burnsac.xyz";

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/burnsac.xyz/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/burnsac.xyz/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
}

server {
    if ($host = www.burnsac.xyz) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    if ($host = burnsac.xyz) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

	listen 80 ;
	listen [::]:80 ;

	server_name burnsac.xyz www.burnsac.xyz;
    return 404; # managed by Certbot
}
```

#### `index.html` - File where the javascript is inserted
- It is within a `</body>` tag and an `</html>` tag
- I tried moving it around, but still to no avail

```
<script data-isso="//isso.burnsac.xyz/burnsac.xyz"
  src="//isso.burnsac.xyz/burnsac.xyz/js/embed.min.js"></script>
  <section id="isso-thread"></section>
</body>
</html>
```

**Comment by stefangehn** [comment]
Check if you see a request ending up in your nginx log when hitting the submit button.
If you don't then check your browser console for errors.

**Comment by lmburns** [comment]
> Check if you see a request ending up in your nginx log when hitting the submit button.
> If you don't then check your browser console for errors.

I do see several entries.

```
IP-ADDRESS - - [27/Feb/2021:20:08:05 +0000] "POST /burnsac.xyz/new?uri=%2F HTTP/2.0" 500 290 "https://burnsac.xyz/" "Mozilla/5.0 (X11; Linux x86_64; rv:83.0) Gecko/2010   0101 Firefox/83.0
```
I'm not sure the issue. Do you know of something else I could check?

**Comment by lmburns** [comment]
Closing this, I tried so much to try and fix this. The issue was that it was attempting to write to a read-only database. When installing `isso` with `pip`, I was unable to specify a database that had not been created yet, and tried changing user permissions but still nothing. However, when installing with Debian, I specified the same file to be the database and saw that after hitting submit on the comment box, the `systemctl status isso` had the error `sqlite3.OperationalError: attempt to write a readonly database`. So I changed the database to a file that did not exist yet and wah lah, it worked--finally!
